plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -15,39 +34,70 @@
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Link;
 import org.eclipse.swt.widgets.Text;
-import org.eclipse.ui.PartInitException;
-import org.eclipse.ui.PlatformUI;
-import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
 import org.eclipse.ui.dialogs.PropertyPage;
 import org.eclipse.ui.forms.widgets.ExpandableComposite;
+import org.eclipse.ui.preferences.ScopedPreferenceStore;
 
 import com.google.cloud.tools.eclipse.ui.util.FontUtil;
+import com.google.cloud.tools.eclipse.ui.util.databinding.BucketNameValidator;
+import com.google.cloud.tools.eclipse.ui.util.databinding.ProjectIdValidator;
+import com.google.cloud.tools.eclipse.ui.util.databinding.ProjectVersionValidator;
+import com.google.cloud.tools.eclipse.ui.util.event.OpenUrlSelectionListener;
+import com.google.cloud.tools.eclipse.util.AdapterUtil;
+import com.google.common.base.Preconditions;
 
 public class DeployPropertyPage extends PropertyPage {
 
   private static final String APPENGINE_DASHBOARD_URL = "https://console.cloud.google.com/appengine";
+
   private static final int INDENT_CHECKBOX_ENABLED_WIDGET = 10;
 
-  private Button promptForProjectIdButton;
+  private static final String PREFERENCE_STORE_DEFAULTS_QUALIFIER =
+      "com.google.cloud.tools.eclipse.appengine.deploy.defaults";
+  private static final String PREFERENCE_STORE_QUALIFIER = "com.google.cloud.tools.eclipse.appengine.deploy";
+  private static final String PREF_PROMPT_FOR_PROJECT_ID = "project.id.promptOnDeploy"; // boolean
+  private static final String PREF_PROJECT_ID = "project.id";
+  private static final String PREF_OVERRIDE_DEFAULT_VERSIONING = "project.version.overrideDefault"; // boolean
+  private static final String PREF_CUSTOM_VERSION = "project.version";
+  private static final String PREF_ENABLE_AUTO_PROMOTE = "project.promote"; // boolean
+  private static final String PREF_OVERRIDE_DEFAULT_BUCKET = "project.bucket.overrideDefault"; // boolean
+  private static final String PREF_CUSTOM_BUCKET = "project.bucket";
```

Comment:
Preferences are a runtime/core thing: they're defined in Equinox (`org.eclipse.equinox.preferences`, incl `IPreferencesService`). `IPreferenceStore` and `ScopedPreferenceStore` are defined in JFace, but `ScopedPreferenceStore` essentially works the same as `IPreferencesService` (consults a set of scopes in order, incl the `DefaultScope`).